app/src/main/java/mozilla/lockbox/presenter/AppRoutePresenter.kt

```diff
@@ -0,0 +1,173 @@
+            is RouteAction.LockScreen -> navigateTo(R.id.fragment_locked)
+            is RouteAction.Filter -> navigateTo(R.id.fragment_filter)
+            is RouteAction.ItemDetail -> navigateTo(R.id.fragment_item_detail, bundle(action))
+            is RouteAction.OpenWebsite -> openWebsite(action.url)
+            is RouteAction.SystemSetting -> openSetting(action)
+            is RouteAction.AutoLockSetting -> showAutoLockSelections()
+            is RouteAction.DialogFragment.FingerprintDialog ->
+                showDialogFragment(FingerprintAuthDialogFragment(), action)
+            is DialogAction -> showDialog(action)
+            is AppWebPageAction -> navigateTo(R.id.fragment_webview, bundle(action))
+        }
+    }
+
+    private fun navigateTo(@IdRes destinationId: Int, args: Bundle? = null) {
+        super.navigateToFragment(navController, destinationId, args)
+    }
+
+    private fun bundle(action: AppWebPageAction): Bundle {
+        return AppWebPageFragmentArgs.Builder()
+            .setUrl(action.url!!)
+            .setTitle(action.title!!)
+            .build()
+            .toBundle()
+    }
+
+    private fun bundle(action: RouteAction.ItemDetail): Bundle {
+        return ItemDetailFragmentArgs.Builder()
+            .setItemId(action.id)
+            .build()
+            .toBundle()
+    }
+
+    private fun showDialog(destination: DialogAction) {
+        AlertDialogHelper.showAlertDialog(activity, destination.viewModel)
+            .map { alertState ->
+                when (alertState) {
+                    AlertState.BUTTON_POSITIVE -> {
+                        destination.positiveButtonActionList
+                    }
+                    AlertState.BUTTON_NEGATIVE -> {
+                        destination.negativeButtonActionList
+                    }
+                }
+            }
+            .flatMapIterable { it }
+            .subscribe(dispatcher::dispatch)
+            .addTo(compositeDisposable)
+    }
+
+    private fun showAutoLockSelections() {
+        val autoLockValues = Setting.AutoLockTime.values()
+        val items = autoLockValues.map { it.stringValue }.toTypedArray()
+
+        settingStore.autoLockTime.take(1)
+            .map { autoLockValues.indexOf(it) }
+            .flatMap {
+                AlertDialogHelper.showRadioAlertDialog(
+                    activity,
+                    R.string.auto_lock,
+                    items,
+                    it,
+                    negativeButtonTitle = R.string.cancel
+                )
+            }
+            .map {
+                autoLockValues[it]
+            }
+            .subscribe {
+                dispatcher.dispatch(SettingAction.AutoLockTime(it))
+            }
+            .addTo(compositeDisposable)
+    }
+
+    // could possibly put this in RoutePresenter? slight differences between these versions
+    private fun showDialogFragment(dialogFragment: DialogFragment?, destination: RouteAction.DialogFragment) {
```

Comment:
There is a slight difference between the `showDialogFragment()` functions in here and in AutofillRoutePresenter. Not sure if they can be generalized and put in the parent

Reply:
They should be able to do that, yes. I think we want to keep the `RoutePresenter` version.